### PR TITLE
add additional security for internal data apis fix for 8678

### PR DESCRIPTION
**Pull request by isharac**
Fix for https://github.com/wso2/product-apim/issues/8678

With this fix
Internal service APIs will be allowed to access by super admin or if the request is for tenant, tenant admins only.

Cross tenant internal data access is restricted with this fix.
However super admin is able to access the data of other tenants' as well.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/ApiPoliciesApiServiceImpl.java [hunk 1]
```diff
@@ -20,58 +20,53 @@
...
-                        fromApiPolicyToApiPolicyListDTO(subscriptionValidationDAO.
-                                getAllApiPolicies(xWSO2Tenant))).build();
-            }
-        } else {
-            if (StringUtils.isNotEmpty(policyName)) {
-                return Response.status(Response.Status.BAD_REQUEST.getStatusCode(),
-                        "X-WSo2-Tenant header is missing.").build();
+                if (StringUtils.isNotEmpty(policyName)) {
+                    return Response.status(Response.Status.BAD_REQUEST.getStatusCode(),
+                            "X-WSo2-Tenant header is missing.").build();
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/ApiPoliciesApiServiceImpl.java:63]: WSO2

File: components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/ApplicationPoliciesApiServiceImpl.java [hunk 2]
```diff
@@ -36,32 +36,43 @@
...
+                if (InternalServiceDataUtil.isUserAuthorizedToTenant(validatedTenantDomain)) {
+                    return Response.ok().entity(InternalServiceDataUtil.
+                            fromApplicationPolicyToApplicationPolicyListDTO(subscriptionValidationDAO.
+                                    getAllApplicationPolicies(validatedTenantDomain))).build();
+                }
             }
         } else {
             if (StringUtils.isNotEmpty(policyName)) {
                 return Response.status(Response.Status.BAD_REQUEST.getStatusCode(),
                         "X-WSo2-Tenant header is missing.").build();
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/ApplicationPoliciesApiServiceImpl.java:64]: WSO2

File: components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/SubscriptionPoliciesApiServiceImpl.java [hunk 3]
```diff
@@ -35,32 +36,45 @@
...
+                if (InternalServiceDataUtil.isUserAuthorizedToTenant(validatedTenantDomain)) {
+                    return Response.ok().entity(InternalServiceDataUtil.
+                            fromSubscriptionPolicyToSubscriptionPolicyListDTO(subscriptionValidationDAO.
+                                    getAllSubscriptionPolicies(validatedTenantDomain))).build();
+                }
             }
         } else {
             if (StringUtils.isNotEmpty(policyName)) {
                 return Response.status(Response.Status.BAD_REQUEST.getStatusCode(),
                         "X-WSo2-Tenant header is missing.").build();
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/SubscriptionPoliciesApiServiceImpl.java:64]: WSO2

File: components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/ApplicationPoliciesApiServiceImpl.java (hunk 2 above)
Comment (isharac) [reply components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/ApplicationPoliciesApiServiceImpl.java]: fixed. Thanks

**Comment by bhathiya** [comment]
@isharac why didn't we push this to 3.2.0?

**Comment by isharac** [comment]
Need to revisit since integration tests are failing.

**Pull request closed without merging**